src/main/java/com/goxr3plus/streamplayer/stream/StreamPlayer.java

```diff
@@ -1235,7 +1235,7 @@ public void setPan(final double fPan) {
 	 */
 	public void setGain(final double fGain) {
 		if (isPlaying() || isPaused() && hasControl(FloatControl.Type.MASTER_GAIN, gainControl))
-			gainControl.setValue((float) (20 * Math.log10(fGain != 0.0 ? fGain : 0.0000)));
```

Comment:
This is wrong here.

Reply:
(fGain != 0.0 ? fGain : 0.0000) can be simplified to (fGain).

0.0 and 0.0000 is the same thing.

if fGain == 0d, the second alternative will be used, which is 0.0, which is fGain
if fGain != 02, the first alternative will be used, which is fGain.
Both paths leads to the value of fGain.

Reply:
Hm i think you are right.